ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh

```diff
@@ -237,31 +233,63 @@ cat > run_test_playbook.yaml <<-EOF
-      stat:
-        path: /dev/nvme0n1
-      register: nvme
+    - name: Create {{ REPO_DIR }} directory if it does not exist
+      ansible.builtin.file:
+        path: "{{ REPO_DIR }}"
+        state: directory
+    - debug:
+        var: CLUSTER_TYPE
+    - name: Customize equinix machine
+      block:
+      - name: Fetch equinix metadata
+        ansible.builtin.uri:
+          url: "https://metadata.platformequinix.com/metadata"
+          return_content: yes
+        register: equinix_metadata
+        until: equinix_metadata.status == 200
+        retries: 5
+        delay: 5
+      - name: Setup working directory for machine type m3.large.x86
+        ansible.builtin.shell: |
+          mkfs.xfs -f /dev/nvme0n1
+          mount /dev/nvme0n1 {{ REPO_DIR }}
+        when: "equinix_metadata.json.plan == 'm3.large.x86'"
+      - name: Setup working directory and swap for machine type c3.medium.x86
+        ansible.builtin.shell: |
+          # c3.medium.x86 has 64GB of RAM which is not enough for most of assisted jobs
+          # we need to mount extra swap space in order over commit memory with libvirt/KVM
+          # the machine has 2x240G disks (one is used for the system) and 2x480GB disks
+
+          # Get disk where / is mounted
+          ROOT_DISK=$(lsblk -o pkname --noheadings --path | grep -E "^\S+" | sort | uniq)
+
+          # Setup the smallest disk available (240GB) as swap
+          SWAP_DISK=$(lsblk -o name --noheadings --sort size --path | grep -v "${ROOT_DISK}" | head -n1)
+          mkswap "${SWAP_DISK}"
+          swapon "${SWAP_DISK}"
+
+          # Setup the largest disk available (480GB) for assisted tests
+          REPO_DISK=$(lsblk -o name --noheadings --sort size --path | grep -v "${ROOT_DISK}" | tail -n1)
+          mkfs.xfs -f "${REPO_DISK}"
+          mount "${REPO_DISK}" "{{ REPO_DIR }}"
+        when: "equinix_metadata.json.plan == 'c3.medium.x86'"
+      when: '"packet" in CLUSTER_TYPE'
+    - name: Create {{ MINIKUBE_HOME }} directory if it does not exist
+      ansible.builtin.file:
+        path: "{{ MINIKUBE_HOME }}"
+        state: directory
     - name: Build config.sh file
       template:
         src: ./config.sh.j2
         dest: /root/config.sh
     - name: Print config file content
       debug:
         msg: "{{ lookup('template', './config.sh.j2').split('\n') }}"
-    - name: Use nvme device if exists
+    - name: Retrieve test-infra sources
       ansible.builtin.shell: |
-        mkfs.xfs -f /dev/nvme0n1
-        mount /dev/nvme0n1 {{ DATA_DIR }}
-      when: nvme.stat.exists
-    - name: Extract test-infra repo archive
-      ansible.builtin.unarchive:
-        src: /root/assisted.tar.gz
-        dest: "{{ REPO_DIR }}"
-        remote_src: yes
-        owner: root
-        group: root
-        mode: 0755
+        podman create --name src "{{ ASSISTED_TEST_INFRA_IMAGE }}"
+        podman cp "src:{{ ASSISTED_TEST_INFRA_SRC_PATH }}" - | tar x --strip-components 1 -C "{{ REPO_DIR }}"
```

Comment:
``podman cp -r`` won't work?

Reply:
there's no `-r` for `podman cp`, but I seems the copy is recursive, I will check :+1:

Reply:
it's much simpler now!